designs/use-selectors-at-cache.md

```diff
@@ -0,0 +1,123 @@
+Filter cache ListWatch using selectors
+===================
+## Motivation
+
+The controller-runtime subscribe to changes using a cache mechanism that get
+updated using kubernetes informers, those informers implement a `ListWatch`
+that specify the namespace and type specified at controller builder time.
+
+So the only possible filtering of that cache is namespace and resource type,
+this means that if some user is interested only on Reconcile some resource with
+specific fields it will still cache all the instances on that type increasing
```

Comment:
```suggestion
The only way to filter this cache is by namespace and resource type. In cases where a controller is only interested in a small subset of objects (for example all pods on a node), this might end up not being efficient enough.
```

Reply:
Done